cpp/src/replace/clamp.cu

```diff
@@ -59,25 +59,21 @@ struct dispatch_clamp {
               rmm::mr::device_memory_resource* mr,
               cudaStream_t stream) {
         using ScalarType = cudf::experimental::scalar_type_t<T>;
-        auto lo_scalar = static_cast<ScalarType const*>(&lo);
-        auto hi_scalar = static_cast<ScalarType const*>(&hi);
-        auto lo_value = lo_scalar->value(stream);
-        auto hi_value = hi_scalar->value(stream);
+        auto lo_scalar = static_cast<ScalarType const&>(lo);
+        auto hi_scalar = static_cast<ScalarType const&>(hi);
+        auto lo_value = lo_scalar.value(stream);
+        auto hi_value = hi_scalar.value(stream);
+        auto output = detail::allocate_like(input, input.size(), mask_allocation_policy::RETAIN, mr, stream);
```

Comment:
Actually, this should be `mask_allocation_policy::NEVER`, otherwise this is redundantly allocating memory that will also be allocated in `copy_bitmask` below.

Reply:
Ahh, yes.